hppc/src/main/templates/com/carrotsearch/hppc/KTypeVTypeWormMap.java

```diff
@@ -0,0 +1,2011 @@
+        int nextOffset = next[hashIndex];
+        //Maybe I should return something else than noValue()
+        if (nextOffset <= 0) {
+            return ~hashIndex;
+        }
+        int entryIndex = searchInChain(key, hashIndex, nextOffset);
+
+        return entryIndex < 0 ? ~entryIndex : entryIndex;
+    }
+
+    @Override
+    public boolean indexExists(int index) {
+        assert (index < 0 || (index >= 0 && index < keys.length));
+
+        // Don't check upper bound just as HPPC do
+        return index >= 0;
+    }
+
+    @Override
+    public VType indexGet(int index) {
+        assert (index >= 0 && index < keys.length);
+
+        return Intrinsics.<VType>cast(values[index]);
+    }
+
+    @Override
+    public VType indexReplace(int index, VType newValue) {
+        assert index >= 0 : "The index must point at an existing key.";
+        assert index < keys.length;
+
+        VType previousValue = Intrinsics.<VType>cast(values[index]);
+        values[index] = newValue;
+        return previousValue;
+    }
+
+    @Override
+    public void indexInsert(int index, KType key, VType value) {
+        assert index < 0 : "The index must not point at an existing key.";
+
+        index = ~index;
+        assert ((next[index]) == 0);
+
+        keys[index] = key;
+        values[index] = value;
+        next[index] = END_OF_CHAIN;
+
+        size++;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sBuilder = new StringBuilder();
+        sBuilder.append('{');
+        boolean firstEntry = true;
+        // Iterate all entries.
+        for (int index = 0, entryCount = 0; entryCount < size; index++) {
+            if (next[index] != 0) {
+                if (firstEntry) {
+                    firstEntry = false;
+                } else {
+                    sBuilder.append(", ");
+                }
+                KType key = Intrinsics.<KType>cast(keys[index]);
+                sBuilder.append(key);
+                sBuilder.append('=');
+                VType value = Intrinsics.<VType>cast(values[index]);
+                sBuilder.append(value);
+                entryCount++;
+            }
+        }
+        sBuilder.append('}');
+        return sBuilder.toString();
+    }
+
+    public int getCapacity() {
+        return next.length;
+    }
+
+    @Override
+    public void ensureCapacity(int expectedElements) {
+        allocateBuffers((int)(expectedElements * 1.0 / 0.76));
+    }
+
+    public boolean allocateBuffers(int capacity) {
+        return allocateBuffers(capacity, false);
+    }
+
+    public boolean allocateBuffers(int capacity, boolean onlyIfEnlarged) {
+        return allocateBuffers(capacity, onlyIfEnlarged, true);
+    }
+
+    public int getLoad() {
+        return (int) (100.0 * size / keys.length);
+    }
+
+    @Override
+    public String visualizeKeyDistribution(int characters) {
+        throw new UnsupportedOperationException("Visualization is not supported in Worm Map");
```

Comment:
I should move it to an optional interface. Although this could be implemented here too.